common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -201,6 +202,28 @@ def get_static_transcript(self, request, transcripts):
                 )
         return response
 
+    @XBlock.json_handler
+    def publish_completion(self, data, dispatch):  # pylint: disable=unused-argument
+        """
+        Entry point for completion for student_view.
+
+        Parameters:
+            data: JSON dict:
+                key: "completion"
+                value: float in range [0.0, 1.0]
+
+            dispatch: Ignored.
+        Return value: JSON response (200 on success, 400 for malformed data)
+        """
+        completion_service = self.runtime.service(self, 'completion')
+        if not completion_service or not completion_service.completion_tracking_enabled():
+            raise JsonHandlerError(404, u"Completion service not available")
+        if not 0.0 <= data['completion'] <= 1.0:
```

Comment:
Good idea.  In python 2, a non-number will just sort outside the range of numbers, but in python 3 it will become an error.  Better not to introduce a possible 500 here.